### DISCUSSION TITLE
Date entered without any separators is changed when auto-formatted to include separators

### DISCUSSION THREAD

**Discussion by jrrdnx** [solspace/craft-freeform#1349] [Feature Requests]
**Describe the bug or issue you're experiencing**

When any Date Separator option except 'None' is used and the date is manually typed in without using the datepicker or any separators, the resulting date in changed to a different date when automatically formatted to include the separators.

**Steps to reproduce**

1. In a Date & Time field (using Type of Date, haven't tested with Both), type in a valid date but without any separators
2. When focus is lost and the date is auto-formatted to include separators, the date is changed to a different date

Examples of before and after with two different dates (01012024 changed to 1/12/2024 and 12312023 changed to 12/12/2023):

<img width="160" alt="Screenshot 2024-01-22 at 09 40 41" src="https://github.com/solspace/craft-freeform/assets/4584671/15a7f2fd-2ee9-4c52-b7d9-636458f67e01">
<img width="188" alt="Screenshot 2024-01-22 at 09 40 50" src="https://github.com/solspace/craft-freeform/assets/4584671/6d6e1803-74c9-404f-85b0-08a03e29ee06">
<img width="221" alt="Screenshot 2024-01-22 at 09 43 17" src="https://github.com/solspace/craft-freeform/assets/4584671/13d1b6f0-f1e4-48fd-9edd-bcced9adbda3">
<img width="184" alt="Screenshot 2024-01-22 at 09 43 22" src="https://github.com/solspace/craft-freeform/assets/4584671/724368e4-a4da-438d-b6cf-bf0b24974d61">

**Expected behavior**

Date should be formatted to include the separators without changing the date entered by the user

**Craft & Plugin Info (please complete the following information):**

- Craft Version: Pro 4.6.1
- Freeform Version: 4.1.13
- Freeform Edition: Pro
- Fresh Install or Upgrade: Fresh install

**Comment by jrrdnx** [comment]
Disabling the datepicker prevents the date from being changed incorrectly, but obviously doesn't format the date automatically so requires the user to make sure this is done manually. Using as a workaround for now.

**Comment by kjmartens** [comment]
Sorry about this, and for the long delay @jrrdnx.

Due to this using Flatpickr, I am not sure we are able to address this on our end. I've noted this though, and we'll attempt to revisit this in the future.
